pkg/geo/geomfn/linestring.go

```diff
@@ -252,3 +253,60 @@ func removePoint(lineString *geom.LineString, index int) (*geom.LineString, erro
 
 	return lineString.SetCoords(coords)
 }
+
+// LineSubString returns a LineString being a substring by start and end
+func LineSubString(g geo.Geometry, start, end float64) (geo.Geometry, error) {
+
+	if start < 0 || start > 1 || end < 0 || end > 1 {
+		return g, pgerror.Newf(pgcode.InvalidParameterValue,
+			"start and and must be within 0 and 1")
+	}
+	if start > end {
+		return g, pgerror.Newf(pgcode.Internal,
+			"2nd arg must be smaller then 3rd arg")
+	}
+
+	if start == end {
+		return LineInterpolatePoints(g, start, false)
+	}
+
+	lineT, err := g.AsGeomT()
+	if err != nil {
+		return g, err
+	}
+	lineString, ok := lineT.(*geom.LineString)
+	if !ok {
+		return g, pgerror.Newf(pgcode.InvalidParameterValue,
+			"first parameter has to be of type LineString")
+	}
+
+	var newCoords, tmpCoords []float64
+	var stop bool
+	for i := 0; i < len(lineString.FlatCoords())-1 && !stop; i += 2 {
+		tmpCoords = append(tmpCoords, lineString.FlatCoords()[i:i+2]...)
+		tmpLineString := geom.NewLineStringFlat(geom.XY, tmpCoords)
+		distance := tmpLineString.Length()
+		if start != 0 && i == 0 {
+			distance = start * lineString.Length()
+		}
+
+		if start != 0 && i != 0 && distance < (start*lineString.Length()) {
```

Comment:
hey -- unfortunately this logic is hard for me to follow without comments (the i/start == 0 logic and special casing makes it a little confusing).  

i've tried to simplify this, take a look and see if this is easier to read for you.
```
startDistance := distance * lineString.length()
endDistance := distance * lineString.length()
for i, coord := range lineString.Coords() {
   currentDistance := geom.NewLineString(geom.XY).SetCoords([]geom.Coord{lineString.Coord(0), lineString.Coord(i))
    // if the current distance exceeds the end distance, find the last point and terminate the loop early.
   if currentDistance >= endDistance {
      // Interpolate point and add to flat coords
      break
   } 
   // If we are past the beginning, check if we already have points in the line string.
   // If this is our first point, interpolate the first point.
   // If we have already added a point, simply add the current coordinate in.
   if currentDistance > startDistance {
      if len(flatCoords) == 0 {
         // Intepolate line from beginning and add point
      } else {
          // addPoint
       }
   }
}
```

Reply:
wow, thanks. that seems clearer than mine.
I'll update it and check it